### ISSUE TITLE
"hidekills" does not properly mask the identity of the oper

### ISSUE THREAD

**Issue by sebastianbarfurth** [inspircd/inspircd#1015]
A kill performed with hidekills active will still pass the performing opers name and host to the user who has been killed.

Here is the output as parsed by IRCCloud:

```
23:19:29 :Snoonet Network Operations KILL aFreshMelonKilled :cosmos.snoonet.org!snoonet/operations-manager/aFreshMelon!Snoonet Network Operations (Killed (Snoonet Network Operations ()))
23:19:29 Closing link: (sid25548@id-25548.brockwell.irccloud.com) [Killed (Snoonet Network Operations ())]
```

You can see that it reveals the opers name and host (in this case myself).

**Comment by SaberUK** [comment]
[Confirmed](https://github.com/inspircd/inspircd/blob/insp20/src/commands/cmd_kill.cpp#L137), but you shouldn't be using spaces in your `hidekills` value or you will break clients

**Comment by aFreshMelon** [comment]
Thanks @SaberUK

**Comment by Adam-** [comment]
The host is passed as part of the kill path, but the nick is hidden. You only see your nick there because it is in your host.

**Comment by aFreshMelon** [comment]
Right. But why is the opers host passed at all? I don't think it's uncommon for staff to have their name contained in their host.

**Comment by SaberUK** [comment]
Probably because some other implementation (Unreal?) did it that way and we copied them.

**Comment by aFreshMelon** [comment]
I'd still love it if it was changed or if there was an option to change it.

**Comment by Adam-** [comment]
I wouldn't mind just removing the kill path from the server to client KILL message. Many modern IRCds do not send it (hybrid, ratbox, charybdis).

**Comment by SaberUK** [comment]
@aFreshMelon Could you test this? https://github.com/SaberUK/inspircd/commit/8a7fdfc4242c

**Comment by aFreshMelon** [comment]
It works, though it does have some weird parenthesis.

```
15:34:01 ⇐ You were killed
15:34:01 Closing link: (sid25548@brockwell.irccloud.com) [Killed (irc.snoonet.org ())]
```

**Comment by attilamolnar** [comment]
Fixed by @SaberUK in #1018.
